kernel-open/nvidia/nv-acpi.c

```diff
@@ -737,6 +737,9 @@ void NV_API_CALL nv_acpi_methods_uninit(void)
 #if defined(NV_ACPI_BUS_GET_DEVICE_PRESENT)
     acpi_bus_get_device(nvif_parent_gpu_handle, &device);
 
+    if (device == NULL)
```

Comment:
Why aren't you using the `ACPI_FAILURE` macro to check the result of `acpi_bus_get_device`?
You could check other places where `acpi_bus_get_device` is being used and do the same.